src/Microsoft.AspNet.Mvc/MvcServices.cs

```diff
@@ -138,6 +139,9 @@ public static IEnumerable<IServiceDescriptor> GetDefaultServices(IConfiguration
             // Only want one ITagHelperActivator so it can cache Type activation information. Types won't conflict.
             yield return describe.Singleton<ITagHelperActivator, DefaultTagHelperActivator>();
 
+            // Consumed by the Cache tag helper to cache results across the lifetime of the application.
+            yield return describe.Singleton<IMemoryCache, MemoryCache>();
```

Comment:
Spoke @loudej and the plan is to share this. We don't need to do anything in particular to have it use `TryAdd` since we get that from using `AddMvc` - https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNet.Mvc/MvcServiceCollectionExtensions.cs#L19